src/main.ts

```diff
@@ -27,19 +39,98 @@ const createWindow = (): void => {
   });
 
   // and load the index.html of the app.
-  mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
+  window.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
 
   // Open the DevTools.
-  mainWindow.webContents.openDevTools();
+  window.webContents.openDevTools();
 
   // Run API
-  runAPI(ipcMain, mainWindow.webContents);
+  runAPI(ipcMain, window.webContents);
+};
+
+// Based on https://cdruc.com/positioning-electron-tray-apps-on-windows-taskbar/
+const getWindowPosition = () => {
+  const trayBounds = tray.getBounds();
+
+  // There may be more than one screen, so we need to figure out on which screen our tray icon lives.
+  const trayScreen = screen.getDisplayNearestPoint({
+    x: trayBounds.x,
+    y: trayBounds.y,
+  });
+
+  // Now that we know the display, we can grab its bounds and its workspace area.
+  const { workArea } = trayScreen;
+  const screenBounds = trayScreen.bounds;
+
+  // TASKBAR LEFT
+  if (workArea.x > 0) {
+    // The workspace starts more on the right
+    return {
+      x: workArea.x,
+      y: workArea.height - HEIGHT,
+    };
+  }
+
+  // TASKBAR TOP
+  if (workArea.y > 0) {
+    return {
+      x: Math.round(trayBounds.x + trayBounds.width / 2 - WIDTH / 2),
+      y: trayBounds.height,
+    };
+  }
+
+  // TASKBAR RIGHT
+  // Here both workArea.y and workArea.x are 0 so we can no longer leverage them. We can use the workarea and display width though.
+  if (workArea.width < screenBounds.width) {
+    // The taskbar is either on the left or right, but since the LEFT case was handled above, we can be sure we're dealing with a right taskbar
+    return {
+      x: workArea.width - WIDTH,
+      y: screenBounds.height - HEIGHT,
+    };
+  }
+
+  // TASKBAR BOTTOM
+  // Since all the other cases were handled, we can be sure we're dealing with a bottom taskbar
+  return {
+    x: Math.round(trayBounds.x + trayBounds.width / 2 - WIDTH / 2),
+    y: workArea.height - HEIGHT,
+  };
+};
+
+const showWindow = () => {
+  const position = getWindowPosition();
+  window.setPosition(position.x, position.y, false);
+  // Added because setPosition would sometimes squeeze the sizing
+  window.setSize(WIDTH, HEIGHT, false);
+  window.show();
+  window.focus();
+};
+
+const toggleWindow = () => {
+  if (window.isVisible()) {
+    window.hide();
+  } else {
+    showWindow();
+  }
```

Comment:
you can use ternary operators here

Reply:
I don't understand how you would use ternaries here, can you explain?

Reply:
```ts
window.isVisible() ? window.hide() : showWindow()
```